docs/advanced/document-store/redis.md

```diff
@@ -0,0 +1,339 @@
+(redis)=
+# Redis
+
+One can use [Redis](https://redis.io) as the document store for DocumentArray. It is useful when one wants to have faster Document retrieval on embeddings, i.e. `.match()`, `.find()`.
+
+````{tip}
+This feature requires `redis`. You can install it via `pip install "docarray[redis]".`
+````
+
+## Usage
+
+### Start Redis service
+
+To use Redis as the storage backend, it is required to have the Redis service started. Create `docker-compose.yml` as follows:
+
+```yaml
+version: "3.3"
+services:
+  redis:
+    image: redislabs/redisearch:2.6.0
+    ports:
+      - "6379:6379"
+```
+
+Then
+
+```bash
+docker-compose up
+```
+
+### Create DocumentArray with Redis backend
+
+Assuming service is started using the default configuration (i.e. server address is `localhost:6379`), one can instantiate a DocumentArray with Redis storage as such:
+
+```python
+from docarray import DocumentArray
+
+da = DocumentArray(storage='redis', config={'n_dim': 128})
+```
+
+The usage would be the same as the ordinary DocumentArray, but the dimension of an embedding for a Document must be provided at creation time.
+
+```{caution}
+Currently, one Redis server instance can only support to store a single DocumentArray.
+```
+
+To store a new DocumentArray in current Redis server, one can set `flush` to `True` so that previous DocumentArray will be cleared:
+
+```python
+from docarray import DocumentArray
+
+da = DocumentArray(storage='redis', config={'n_dim': 128, 'flush': True})
+```
+
+To access a previously stored DocumentArray, one can specify `host` and `port`.
```

Comment:
I think this is not accurate, host a port are mostly about the target instance. You can still not specify host and port and have access to a previously stored document array. Maybe you're referring to flush = False ?